zebra-consensus/src/errors.rs

```diff
@@ -0,0 +1,44 @@
+//! Consensus validation errors
+
+use thiserror::Error;
+
+#[derive(Error, Debug)]
+pub enum TransactionError {
+    #[error("no transactions")]
+    NoTransactions,
+
+    #[error("first transaction must be coinbase")]
+    CoinbasePosition,
+
+    #[error("coinbase input found in non-coinbase transaction")]
+    CoinbaseInputFound,
+}
```

Comment:
My preference would be to use displaydoc here (I'm pretty biased).
```suggestion
#[derive(Error, Debug, Display)]
pub enum TransactionError {
    /// no transactions
    NoTransactions,
    /// first transaction must be coinbase
    CoinbasePosition,
    /// coinbase input found in non-coinbase transaction
    CoinbaseInputFound,
}
```

Reply:
I think it is most consistent with `BlockError` as it is but i dont have any strong preference. Can change if you guys think it is better.